core/src/main/java/com/amplitude/core/Amplitude.kt

```diff
@@ -68,104 +73,131 @@ open class Amplitude internal constructor(
     }
 
     @Deprecated("Please use 'track' instead.", ReplaceWith("track"))
-    fun logEvent(event: BaseEvent) {
-        track(event)
+    fun logEvent(event: BaseEvent): Amplitude {
+        return track(event)
     }
 
-    fun track(event: BaseEvent, callback: EventCallBack? = null) {
+    /**
+     * Track an event
+     */
+    fun track(event: BaseEvent, callback: EventCallBack? = null): Amplitude {
         callback ?. let {
             event.callback = it
         }
         process(event)
+        return this
     }
 
-    fun track(eventType: String, eventProperties: JSONObject? = null, options: EventOptions? = null) {
+    /**
+     * Log event with the specified event type, event properties, and optional event options
+     */
+    fun track(eventType: String, eventProperties: JSONObject? = null, options: EventOptions? = null): Amplitude {
         val event = BaseEvent()
         event.eventType = eventType
         event.eventProperties = eventProperties
         options ?. let {
             event.mergeEventOptions(it)
         }
         process(event)
+        return this
     }
 
     /**
      * Identify. Use this to send an Identify object containing user property operations to Amplitude server.
```

Comment:
Should we explain what exactly this does? 
Like - 
You can modify user properties by calling this API.

Reply:
Add that in the comments